### PR TITLE
[REVIEW] ORC reader: add option to read DECIMALs as INT64

### PR DESCRIPTION
**Pull request by OlivierNV**
- [x] Add decimals-as-int support in ORC reader kernels
- [x] Add CPP API for decimals-as-float and forced decimal scale
- [x] Add new parameters to python api
- [x] Add simple python test

ORC part of #3306

### CODE REVIEW

**Comment by OlivierNV** [comment]
rerun tests

**Comment by OlivierNV** [comment]
rerun tests

**Comment by mjsamoht** [comment]
@randerzander can you review the API change to support integer fixed point?  Is that what you had in mind?

**Comment by mjsamoht** [comment]
> @randerzander can you review the API change to support integer fixed point? Is that what you had in mind?

I spoke to @randerzander this afternoon, and he was okay with the proposed API.

**Comment by OlivierNV** [comment]
The libcudf++ path does not have bindings yet ~ python tests could be added for the legacy path, but there is no pandas equivalent, and we can't write decimals in the writer for CPP tests. (I only verified offline that the int values generated matched expected results when reading the existing ORC file with decimals).

**Comment by OlivierNV** [comment]
Added a simple test (using python+legacy path) that checks the first value of the TestOrcFile.Decimals.orc test file.

**Comment by harrism** [comment]
Strange CI failure seems to be one we were seeing last week. Suggest merging latest branch-0.12 and rerunning, since a couple of hot fixes for CI failures went into 0.11 this morning.

File: cpp/include/cudf/legacy/io_readers.hpp [hunk 1]
```diff
@@ -322,11 +324,14 @@ struct reader_options {
    * @param[in] timestamp_time_unit Resolution of timestamps; none for default
    *---------------------------------------------------------------------------**/
   reader_options(std::vector<std::string> cols, bool use_index_lookup,
-                 bool np_compat, gdf_time_unit timestamp_time_unit)
+                 bool np_compat, gdf_time_unit timestamp_time_unit,
+                 bool decimals_as_float_ = true, int forced_decimals_scale_ = -1)
```
[the remark this answers is not in the document]
Comment (OlivierNV) [reply cpp/include/cudf/legacy/io_readers.hpp]: The other parameters are using different names than the member (decimals_as_float = decimals_as_float would be ambiguous afaik)

**Comment by OlivierNV** [comment]
Cuda9.2 still hit the following failures
7 - LEGACY_HASH_GROUPBY_TEST (Failed)
8 - LEGACY_SORT_GROUPBY_TEST (Failed)
31 - LEGACY_BINARY_TEST (Failed)
rerun tests.

**Pull request merged**